### DISCUSSION TITLE
horizontal buttons Card

### DISCUSSION THREAD

**Discussion by telegrambeans** [Clooos/Bubble-Card#495] [Questions about config, custom styles and templates] [answered]
How can I animate the icon with state? 

Thx

**Comment by Clooos** [comment]
Hi, do you have a more specific example? I'm not sure to understand what you want to achieve.

**Comment by telegrambeans** [comment]
I want to animate the icon of the horizontal buttons card depending on the status. E.g. room 1 is occupied and the symbol is moving:

```
animation: beat 1.3s ease-out infinite both;
      transform-origin: 50% 60%;

@keyframes beat {
      0% { transform: scale(1); }
      10% { transform: scale(1.1); }
      17% { transform: scale(1.05); }
      33% { transform: scale(1.25); }
      60% { transform: scale(1); }
    }
```

**Comment by telegrambeans** [comment]
It only works with an animation, but only because it is present in the source code. So he doesn't use the keyframe. 

Example:

```
  .wohnzimmer {
    background: transparent !important;
    border: transparent !important;
    ha-icon {
     animation: ${hass.states['binary_sensor.bewegung_wohnzimmer_gesamt'].state === 'on' ? 'pulse 2s infinite alternate' : 'none'}! important;
     @keyframes pulse {
        0% {filter: brightness(0.1) !important;}
        50% {filter: brightness(0.5) !important;}
        100% {filter: brightness(1.0) !important;}
        50% {filter: brightness(0.5) !important;}
        0% {filter: brightness(0.1) !important;}
      }
     color: ${hass.states['light.stehlampe_l'].state === 'on' ? '#fdc10a' : 'withe'}! important;
    } 
  }
```

**Comment by telegrambeans** [comment]
Example icon:


https://github.com/Clooos/Bubble-Card/assets/13040847/c7560599-53c5-4809-9c06-33d3b6663ec9

**Comment by telegrambeans** [comment]
Is there a way to do that?

**Comment by Clooos** [comment]
I guess but I'm not sure, I will have to try that but it's not my priority right now because I have a lot of other issues 😅 

But if you progress on that please tell me how you did it.

**Comment by pgale** [comment, marked as answer]
I know this is an old-ish post. I couldn't get the icon animation working through the custom styles section of Bubble Card but you can do it with card_mod. This is how I animated a gentle rotation for a fan switch:

```
card_mod:
  style: |
    .bubble-icon {
      animation: {{ 'slow-rotate 3s linear infinite' if states('light.bedroom_fan') == 'on' else 'none' }};
    }
    @keyframes slow-rotate {
      0% { transform: rotate(0deg); }
      100% { transform: rotate(360deg); }
    }
```

This how it would look like in the full card:

```
- type: custom:bubble-card
  card_type: button
  button_type: switch
  entity: light.bedroom_fan
  name: Bedroom fan
  show_state: false
  show_attribute: false
  tap_action:
    action: toggle
  double_tap_action:
    action: none
  hold_action:
    action: none
  card_layout: large
  card_mod:
    style: |
      .bubble-icon {
        animation: {{ 'slow-rotate 3s linear infinite' if states('light.bedroom_fan') == 'on' else 'none' }};
      }
      @keyframes slow-rotate {
        0% { transform: rotate(0deg); }
        100% { transform: rotate(360deg); }
      }
  icon: mdi:fan
```

You can setup whatever animation takes your fancy. For example I have a custom alarm clock I created with a load of helpers and an automation. When the alarm is snoozed, the alarm on/off switch icon shakes and rotates back and forth like a cartoon alarm clock moving.

@Clooos might be worth adding to the docs unless you plan on supporting the animation style?

**Reply by Clooos** [reply]
There is no need to use card_mod for that, this is working with the Bubble Card custom styles:

```yaml
styles: |-
  .bubble-icon {
    animation: ${hass.states['fan.you_fan'].state === 'on' ? 'slow-rotate 2s linear infinite' : ''};
  }
  @keyframes slow-rotate {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
  }
```

And for an icon in the horizontal buttons stack:

```yaml
.kitchen > .bubble-icon {
  animation: ${hass.states['fan.you_fan'].state === 'on' ? 'slow-rotate 2s linear infinite' : ''};
}
@keyframes slow-rotate {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}
```

**Reply by pgale** [reply]
ok - I didn't see it in your docs - did I miss it?

**Reply by pgale** [reply]
and is that the same in switches on a dashboard AND in a popup?

**Reply by Clooos** [reply]
No it was not there, but this is basically the same as the other conditional examples.

I will add this one in the doc.

**Reply by pgale** [reply]
> No it was not there, but this is basically the same as the other conditional examples.
> 
> I will add this one in the doc.

That will help :)

**Reply by Clooos** [reply]
> and is that the same in switches on a dashboard AND in a popup?

Yes 🙂

**Reply by pgale** [reply]
Just replaced them all - working nicely :)

**Reply by Clooos** [reply]
Done 👌

<img width="869" alt="image" src="https://github.com/Clooos/Bubble-Card/assets/36499953/5cb4e304-39ec-4691-908b-b6e6c03a6403">

**Reply by pgale** [reply]
Nice :)
